src/phasorpy/phasor.py

```diff
@@ -0,0 +1,507 @@
+    ...     known_phase, known_modulation
+    ... )
+    >>> phase0
+    0.2
+    >>> modulation0
+    3.25
+    """
+    measured_phase = numpy.asarray(measured_phase)
+    measured_modulation = numpy.asarray(measured_modulation)
+    known_phase = numpy.asarray(known_phase)
+    known_modulation = numpy.asarray(known_modulation)
+    phase0 = measured_phase - known_phase
+    modulation0 = measured_modulation / known_modulation
+    return phase0, modulation0
+
+
+def phasor_to_polar(
+    real: ArrayLike,
+    imag: ArrayLike,
+    /,
+) -> tuple[NDArray[Any], NDArray[Any]]:
+    """Return polar coordinates from phasor coordinates.
+
+    Parameters
+    ----------
+    real : array_like
+        Scalar or array containing the real components of the
+        phasor coordinates.
+    imag : array_like
+        Scalar or array containing the imaginary components of the
+        phasor coordinates.
+
+    Returns
+    -------
+    tuple[NDArray[Any], NDArray[Any]]
+        phase:
+            Phase values calculated from the phasor coordinates.
+        modulation:
+            Modulation values calculated from the phasor coordinates.
+
+    Examples
+    --------
+    >>> real_data = numpy.array([1.0, 2.0, 3.0])
+    >>> imag_data = numpy.array([4.0, 5.0, 6.0])
+    >>> phase, modulation = phasor_to_polar(
+    ...     real_data, imag_data
+    ... )
+    >>> phase
+    array(...)
+    >>> modulation
+    array(...)
+    """
+    real = numpy.asarray(real)
+    imag = numpy.asarray(imag)
+    phase = numpy.arctan2(imag, real)
+    modulation = numpy.hypot(real, imag)
+    return phase, modulation
+
+
+def phasor_from_lifetime(
+    frequency: float,
+    lifetime: ArrayLike,
+    /,
+    fraction: ArrayLike | None = None,
+    *,
+    is_preexp: bool = False,
+) -> tuple[NDArray[numpy.float64], NDArray[numpy.float64]]:
+    """Return phasor coordinates from multi-exponential lifetime components.
+
+    Parameters
+    ----------
+    frequency: float
+        Laser pulse or modulation frequency in MHz.
+    lifetime: array_like
+        Lifetime components in ns. May be a scalar for single-exponential
+        or a sequence for multi-exponential components.
+    fraction: array_like, optional
+        Fractional intensities or pre-exponential amplitudes of the lifetime
+        components. Must be of same shape as `lifetime`.
+        If None (default), fractions are 1 for all lifetime components.
+        Fractions are normalized internally.
+    is_preexp: bool, optional
+        If true, `fraction` values are pre-exponential amplitudes,
+        else fractional intensities (default).
+
+    Returns
+    -------
+    real, imag: ndarray
+        Real and imaginary components of phasor coordinates.
+
+    Raises
+    ------
+    ValueError
+        Input arrays do not match or are more than one dimensional.
+
+    Examples
+    --------
+    >>> phasor_from_lifetime(80.0, 1.9894368)  # 1000/(2*pi*80) = 1.9894368
+    (0..., 0...)
```

Comment:
Yes I agree. What was easiest was to use values that give more "round" returned values so that we don't have that problem for decimal precision, but I know this might not be possible in all cases. I did a quick research but couldn't find anything for this problem. Maybe we can add an Issue so that we can remember to try find a solution later?

Reply:
Well, in this case the expected values are `(0.5, 0.5)`. It does not get much rounder than that. Unfortunately the returned values are `0.49999999` or similar due to floating point processing... One option would be to mark the examples with `# doctest: +SKIP`, another option to explicitly round or print the returned values with a certain precision. Both options are not nice for docstring examples. Maybe setting [numpy.set_printoptions](https://numpy.org/doc/stable/reference/generated/numpy.set_printoptions.html) works?

Reply:
We have to find a solution for this sooner than later.

Reply:
Turned out I misunderstood the [pytest documentation](https://docs.pytest.org/en/7.1.x/how-to/doctest.html#using-doctest-options). Marking the examples with `# doctest: +NUMBER` works as we want:
```
    >>> phasor_from_lifetime(80.0, 1.9894368)  # doctest: +NUMBER
    (0.5, 0.5)
```